qos-service-impl/src/main/java/com/palantir/atlasdb/qos/QosResource.java

```diff
@@ -16,20 +16,77 @@
 
 package com.palantir.atlasdb.qos;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
+import com.google.common.primitives.Ints;
+import com.palantir.atlasdb.qos.config.CassandraHealthMetricMeasurement;
+import com.palantir.atlasdb.qos.config.ImmutableCassandraHealthMetricMeasurement;
+import com.palantir.atlasdb.qos.config.ImmutableQosClientLimitsConfig;
+import com.palantir.atlasdb.qos.config.QosCassandraMetricsConfig;
+import com.palantir.atlasdb.qos.config.QosClientLimitsConfig;
+import com.palantir.atlasdb.qos.config.QosPriority;
 import com.palantir.atlasdb.qos.config.QosServiceRuntimeConfig;
+import com.palantir.cassandra.sidecar.metrics.CassandraMetricsService;
+import com.palantir.remoting3.clients.ClientConfigurations;
+import com.palantir.remoting3.jaxrs.JaxRsClient;
 
 public class QosResource implements QosService {
 
+    private final Optional<CassandraMetricsService> cassandraMetricClient;
     private Supplier<QosServiceRuntimeConfig> config;
 
     public QosResource(Supplier<QosServiceRuntimeConfig> config) {
```

Comment:
i think the config object is a little too much to be passing into this class. Could we maybe extract a class responsible for returning configured limits for a client, and another for calculating the multiplier?